File: Algoritmica/Actividades/Actividad-IV/Soluciones/Ejercicio-7/Ejercicio-7.py
from Class_MonticuloMax import *

def imprimir_monticulo_maximo(monticulo):
    print("Montículo máximo construido:", monticulo.listaMonticulo[1:])

# Crear un montículo máximo y probar los métodos
monticulo_maximo = MonticuloMaximo()
lista_desordenada = [9, 5, 7, 2, 3, 6, 1, 8, 4]

# Construir el montículo máximo
monticulo_maximo.construir_monticulo(lista_desordenada)

# Imprimir el montículo máximo
imprimir_monticulo_maximo(monticulo_maximo)

# Eliminar y imprimir el máximo repetidamente para ordenar la lista
lista_ordenada = []
while monticulo_maximo.tamanoActual > 0:
    lista_ordenada.append(monticulo_maximo.eliminar_maximo())
print("Lista ordenada:", lista_ordenada)

File: Algoritmica/Actividades/Actividad-IV/Soluciones/Ejercicio-7/Ejercicio-7.cpp
#include <bits/stdc++.h>
#include "Class_MonticuloMax.hpp"

using namespace std;

void imprimir_monticulo_maximo(const MonticuloMaximo& monticulo) {
    cout << "Montículo máximo construido: ";
    for (int i = 1; i <= monticulo.tamanoActual; i++) {
        cout << monticulo.listaMonticulo[i] << " ";
    }
    cout << endl;
}

int main() {
    // Crear un montículo máximo y probar los métodos
    MonticuloMaximo monticulo_maximo;
    vector<int> lista_desordenada = {9, 5, 7, 2, 3, 6, 1, 8, 4};

    // Construir el montículo máximo
    monticulo_maximo.construir_monticulo(lista_desordenada);

    // Imprimir el montículo máximo
    imprimir_monticulo_maximo(monticulo_maximo);

    // Eliminar y imprimir el máximo repetidamente para ordenar la lista
    cout << "Lista ordenada: ";
    while (monticulo_maximo.getTamanoActual() > 0) {
        cout << monticulo_maximo.eliminar_maximo() << " ";
    }
    cout << endl;

    return 0;
}
